bugsnag-android-performance/src/test/java/com/bugsnag/android/performance/internal/ImmutableConfigTest.kt

```diff
@@ -177,6 +179,44 @@ class ImmutableConfigTest {
         verify(logger).w(startsWith("Invalid configuration"))
     }
 
+    @Test
+    fun EditSpandEndCallbacks() {
+        val perfConfig = PerformanceConfiguration(mockedContext(), TEST_API_KEY).apply {
+            endpoint = "https://test.com/testing"
+            releaseStage = "staging"
+            enabledReleaseStages = setOf("staging", "production")
+            autoInstrumentAppStarts = false
+            autoInstrumentActivities = AutoInstrument.START_ONLY
+            versionCode = 543L
+            appVersion = "9.8.1"
+            tracePropagationUrls = emptySet<Pattern>()
+        }
+
+        val spanEndCallback1 = SpanEndCallback { span: Span ->
+            span as SpanImpl
+            span.setAttribute("Test Number", 1111)
+            true
+        }
+        val spanEndCallback2 = SpanEndCallback { span: Span ->
+            span as SpanImpl
+            span.setAttribute("Test Number", 2222)
+            true
+        }
+        val spanEndCallback3 = SpanEndCallback { span: Span ->
+            span as SpanImpl
+            span.setAttribute("Test Number", 3333)
+            true
+        }
```

Comment:
The logic in these doesn't appear to be used at all, maybe they should all rather be the same no-op callback?